src/main/kotlin/org/rust/ide/annotator/cargoCheck/RsCargoCheckUtils.kt

```diff
@@ -320,15 +317,21 @@ private fun formatMessage(message: String): String {
  * these documents should be stripped later (when [saveAllDocuments] is called).
  */
 private fun saveAllDocumentsAsTheyAre() {
+    checkWriteAccessAllowed()
     val documentManager = FileDocumentManager.getInstance()
-    for (document in documentManager.unsavedDocuments) {
+    val rustfmtWatcher = ApplicationManager.getApplication().getComponent(RustfmtWatcher::class.java)
+    val rustfmtWatcherStatus = rustfmtWatcher.isEnabled
+    rustfmtWatcher.isEnabled = false
+    val (pre, post) = documentManager.unsavedDocuments.partition { it.virtualFile?.name in Rustfmt.CONFIG_FILES }
+    for (document in pre + post) {
         documentManager.saveDocumentAsIs(document)
         documentManager.stripDocumentLater(document)
+        rustfmtWatcher.reformatDocumentLater(document)
     }
+    rustfmtWatcher.isEnabled = rustfmtWatcherStatus
```

Comment:
I'd prefer to wrap all code between changing `rustfmtWatcher.isEnabled` value to `try/finally` block to be sure that the initial value is restored even in case of exception